### ISSUE TITLE
Several draw methods do not use the default color palette

### ISSUE THREAD

**Issue by josephmure** [openturns/openturns#1779]
For example, [distribution](https://openturns.github.io/openturns/1.16/user_manual/_generated/openturns.Normal.html) `draw` methods.

**Comment by jschueller** [comment]
this is ok for me for single curves, the palette should be used when drawing a collection of items

**Comment by josephmure** [comment]
Could you please reopen this issue @jschueller? I would like to know the opinion of @KieranDelamotte @mbaudin47.

**Comment by mbaudin47** [comment]
I do not think that the current is ok for single curves, as they should use the default palette (they are currently using either blue, or red colors): this is meant by the word "default" i.e. used whenever no particular setting was used. 

Moreover, the current default palette seems to be still unused in many places, e.g. in contour plots:

http://openturns.github.io/openturns/master/auto_data_analysis/distribution_fitting/plot_asymptotic_estimators_distribution.html#sphx-glr-auto-data-analysis-distribution-fitting-plot-asymptotic-estimators-distribution-py

![image](https://user-images.githubusercontent.com/31351465/112547525-25f2c500-8dbb-11eb-8448-de63a4991931.png)

In convergence plots:
http://openturns.github.io/openturns/master/auto_reliability_sensitivity/central_dispersion/plot_expectation_simulation_algorithm.html#sphx-glr-auto-reliability-sensitivity-central-dispersion-plot-expectation-simulation-algorithm-py
![image](https://user-images.githubusercontent.com/31351465/112547619-43279380-8dbb-11eb-872c-97ebcdba2912.png)

In mesh plots:
http://openturns.github.io/openturns/master/auto_probabilistic_modeling/stochastic_processes/plot_create_mesh.html#sphx-glr-auto-probabilistic-modeling-stochastic-processes-plot-create-mesh-py
![image](https://user-images.githubusercontent.com/31351465/112547694-63efe900-8dbb-11eb-83e2-2bf334de528e.png)

In Sobol' indices plots:
http://openturns.github.io/openturns/master/auto_reliability_sensitivity/sensitivity_analysis/plot_functional_chaos_sensitivity.html#sphx-glr-auto-reliability-sensitivity-sensitivity-analysis-plot-functional-chaos-sensitivity-py
![image](https://user-images.githubusercontent.com/31351465/112547782-81bd4e00-8dbb-11eb-8113-e5b89fe5726e.png)

Finally, a very useful improvement would be to use the palette by default so as the colors of different objects gets different, if not specifically set by the user.

For example, the following script:
```
import openturns as ot
import openturns.viewer as otv

model = ot.SymbolicFunction(["x1"], ["10.0 + x1 + x1^2 / 20 + 2.0 * x1 * sin(x1)"])
exact_x_min = 10.9911
exact_y_min = 5.04933

graph = model.draw(0.0, 12.0, 100)
cloud = ot.Cloud([exact_x_min], [exact_y_min])
cloud.setPointStyle("circle")
graph.add(cloud)
otv.View(graph)
```
currently produces:

![image](https://user-images.githubusercontent.com/31351465/112549179-a9adb100-8dbd-11eb-94fd-e011a7b01262.png)

This is why I insert the code:
```
graph.setColors(ot.Drawable_BuildDefaultPalette(2))
```
which produces:
![image](https://user-images.githubusercontent.com/31351465/112549249-c6e27f80-8dbd-11eb-9897-adf525f82e4f.png)

This line is in all my scripts, which seems as a defect for me, since Matplotlib does this *by default*.
The way I would proceed would be to add a new boolean attribute `isColorSet` to `False` by default. Then at Viewer time, I would quiery this attribute: if all attributes of all curves in the `Graph` are False, then I would set the colors to the default palette and set them to True.

**Comment by mbaudin47** [comment]
Does the PR #2422 fixes that issues?
